abyss_filters/lvars_alias.py

```diff
@@ -4,12 +4,59 @@
 VAR_ASG_VAR_SUFFIX = "_"
 VAR_ASG_CALL_PREFIX = "result_"
 
+# mapping a type string representation to a deterministic name to use
+VAR_ASG_CALL_DETERMINISTIC = {
+    "NTSTATUS": "status",
+}
+
+fDebug = False
+def debug(msg):
+    if fDebug:
+        print(msg)
+
+def debug_lvars(lvars):
+    if not fDebug:
+        return
+    print("%d variables" % len(lvars))
+    for one in lvars:
+        print("%s %s; // %s" % (one.type(), one.name, one.cmt))
+
+def set_var_unique_name(var_x, var_y, lvars):
+    old_name = var_x.name
+    new_name = var_y.name + VAR_ASG_VAR_SUFFIX
+    new_name = set_unique_name(var_x, new_name, lvars)
+    debug("Renamed: %s (%s) = %s " % (old_name, new_name, var_y.name))
+
+def set_func_unique_name(var_x, func_name, lvars):
+    old_name = var_x.name
+    x_type = var_x.type()
+    if str(x_type) in VAR_ASG_CALL_DETERMINISTIC.keys():
+        new_name = VAR_ASG_CALL_DETERMINISTIC[str(x_type)]
+    else:
+        new_name = VAR_ASG_CALL_PREFIX + func_name
+    new_name = set_unique_name(var_x, new_name, lvars)
+    debug("Renamed: %s (%s) = %s(...) " % (old_name, new_name, func_name))
+
+def set_unique_name(var_x, new_name, lvars):
+    bExist = True
+    while bExist:
+        bExist = False
+        for one in lvars:
+            if new_name == one.name:

```

Comment:
I think this function should be called recursively. Otherwise, there is no guarantee for the new name to be unique.

Reply:
It will add several `VAR_ASG_VAR_SUFFIX` if needed, due to the `while` loop. So will try one at a time until it succeeds. So should be ok afaict (at least from my tests)

Reply:
Yes, I do agree now. I misread your code initially. Thanks :)